pkgs/games/fteqw/default.nix

```diff
@@ -0,0 +1,98 @@
+{ lib
+, stdenv
+, pkgs
+, fetchsvn
+, runCommand
+, pkgName ? "fteqw"
+}:
+let
+  releaseFile = {
+    fteqw = pkgName;
+    fteqw-dedicated = "fteqw-sv";
+    fteqcc = pkgName;
+  };
+in stdenv.mkDerivation rec {
+  pname = pkgName;
+  version = "unstable-2022-08-09";
+  src = fetchsvn {
+    url = "https://svn.code.sf.net/p/fteqw/code/trunk";
+    rev = "6303";
+    sha256 = "sha256-tSTFX59iVUvndPRdREayKpkQ+YCYKCMQe2PXZfnTgPQ=";
+  };
+  makeFlags = [
+    "PKGCONFIG=pkg-config"
+    "-C engine"
+  ];
+  buildFlags = lib.optionals (pkgName == "fteqw") [
+    "m-rel"
+  ] ++ lib.optionals (pkgName == "fteqw-dedicated") [
+    "sv-rel"
+  ] ++ lib.optionals (pkgName == "fteqcc") [
+    "qcc-rel"
+  ];
+  buildInputs = with pkgs; lib.optionals (pkgName == "fteqw") [
+      gzip libvorbis libmad SDL2 SDL2_mixer
+      libpng alsa-lib gnutls libjpeg vulkan-loader
+      speex libopus xorg.libXrandr xorg.libXcursor
+  ] ++ lib.optionals (pkgName == "fteqw-dedicated") [
+      gnutls zlib
+  ] ++ lib.optionals (pkgName == "fteqcc") [
+      zlib
+  ];
+  nativeBuildInputs = with pkgs; lib.optionals (pkgName == "fteqw") [
+      SDL2.dev pkg-config libpng.dev gnutls.dev
+      xorg.libXrandr.dev vulkan-loader.dev vulkan-headers
+  ] ++ lib.optionals (pkgName == "fteqw-dedicated") [
+      gnutls.dev pkg-config zlib.dev
+  ] ++ lib.optionals (pkgName == "fteqcc") [
+      zlib.dev pkg-config
+  ];
+  enableParallelBuilding = true;
+  preBuild = ''
+      substituteInPlace ./engine/Makefile \
+        --replace "I/usr/include/opus" "I${pkgs.libopus.dev}/include/opus"
+      substituteInPlace ./engine/gl/gl_vidlinuxglx.c \
+        --replace 'Sys_LoadLibrary("libXrandr"' 'Sys_LoadLibrary("${pkgs.xorg.libXrandr}/lib/libXrandr.so"'
+    '';
+
+  installPhase = ''
+      mkdir -p $out/bin
+      runHook preInstall
+
+      install -m755 engine/release/${releaseFile.${pkgName}} $out/bin/${pkgName}
+
+      runHook postInstall
+    '';
+
+  postFixup = lib.optionalString (pkgName == "fteqw") ''
+      patchelf \
+        --add-needed ${pkgs.SDL2}/lib/libSDL2.so \
+        --add-needed ${pkgs.libGL}/lib/libGLX.so \
+        --add-needed ${pkgs.libGL}/lib/libGL.so \
+        --add-needed ${pkgs.gnutls.out}/lib/libgnutls.so \
+        --add-needed ${pkgs.vulkan-loader}/lib/libvulkan.so \
+        $out/bin/${pkgName}
+    '' + lib.optionalString (pkgName == "fteqw-dedicated") ''
+      patchelf \
+        --add-needed ${pkgs.gnutls.out}/lib/libgnutls.so \
+        $out/bin/${pkgName}
+    '';
+
+  meta = with lib; {
+    homepage = "https://fte.triptohell.info";
+    description =
+      lib.optionalString (pkgName == "fteqw") "A hybrid and versatile game engine" +
+      lib.optionalString (pkgName == "fteqw-dedicated") "Dedicated server for FTEQW" +
+      lib.optionalString (pkgName == "fteqcc") "User friendly QuakeC compiler";
```

Comment:
This would also move into the common function or file. I imagine something like:

```
  meta = common.meta // {
    description = "A hybrid and versatile game engine";
  };
```